src/btree/bt_ret.c

```diff
@@ -94,19 +94,29 @@ __wt_value_return_buf(WT_CURSOR_BTREE *cbt, WT_REF *ref, WT_ITEM *buf)
     if (page->type == WT_PAGE_ROW_LEAF) {
         rip = &page->pg_row[cbt->slot];
 
-        /* Simple values have their location encoded in the WT_ROW. */
+        /*
+         * Simple values have their location encoded in the WT_ROW.
+         *
+         * Cannot check visibility here so there is chance some visibility issues not caught.
+         */
         if (__wt_row_leaf_value(page, rip, buf))
             return (0);
 
         /* Take the value from the original page cell. */
         __wt_row_leaf_value_cell(session, page, rip, NULL, &unpack);
+
```

Comment:
Not sure if we need to check and return not found or onpage value here. From my understanding of the current search flow, we don't need to ATM, but the flow itself may need to be changed.

Reply:
@keithbostic Can you elaborate on that?